src/trackinfoobject.cpp

```diff
@@ -783,79 +781,85 @@ float TrackInfoObject::getCuePoint() {
 }
 
 void TrackInfoObject::slotCueUpdated() {
-    setDirty(true);
+    markDirty();
     emit(cuesUpdated());
 }
 
-Cue* TrackInfoObject::addCue() {
-    //qDebug() << "TrackInfoObject::addCue()";
+CuePointer TrackInfoObject::addCue() {
     QMutexLocker lock(&m_qMutex);
-    Cue* cue = new Cue(m_id);
```

Comment:
Smart developers use smart pointers ;-) 

I just did not find the place where they where deleted in your patch. Is it still there?